faq.md

```diff
@@ -22,7 +22,17 @@ https://github.com/duckdb/duckdb-web). Pull requests to fix issues or generally
 ### I benchmarked DuckDB and its slower than \[some other system\]
 In a departure from traditional academic systems research practise, we have at first focused our attention on correctness, not raw performance. So it is entirely possible DuckDB is slower than some other, more mature system at this point. That being said, we are now confident DuckDB produces correct query results, and are actively working to make it fast, too. So publishing benchmark numbers from the current preview releases is certainly interesting, but should not be taken as the definitive results on what the DuckDB architecture can or cannot do.
 
-### Does DuckDB use SIMD
+### Does DuckDB use SIMD?
 DuckDB does not use *explicit SIMD* instructions because they greatly complicate portability and compilation. Instead, DuckDB uses *implicit SIMD*, where we go to great lengths to write our C++ code in such a way that the compiler can *auto-generate SIMD instructions* for the specific hardware. As an example why this is a good idea, porting DuckDB to the new Apple M1 architecture took 10 minutes.
 
+### How does DuckDB handle concurrency?
+DuckDB has 2 configurable options for concurrency. 1. One process can both read and write to the database. 2. Multiple processes can read from the database, but no processes can write ([`access_mode = 'READ_ONLY'`](/docs/sql/configuration#configuration-reference)). When using option 1, DuckDB does support multiple writer threads using a combination of MVCC (Multi-Version Concurrency Control) and optimistic concurrency control (see below), but all within that single writer process. The reason for this concurrency model is to allow for the caching of data in RAM for faster analytical queries, rather than going back and forth to disk during each query. It also allows the caching of functions pointers, the database catalog, and other items so that subsequent queries on the same connection are faster. DuckDB is also optimized for bulk operations, so executing many small transactions is not a primary design goal. 
 
+### How does DuckDB handle concurrency within a single process?
+DuckDB supports multiple writer threads using a combination of MVCC (Multi-Version Concurrency Control) and optimistic concurrency control. As long as there are no write conflicts, multiple concurrent writes will succeed. Appends will never conflict, even on the same table. Multiple threads can also simultaneously update separate tables or separate subsets of the same table. Each subset is ~20,000-80,000 rows depending on the width of the rows. Optimistic concurrency control comes into play when two threads attempt to edit (update or delete) the same row of data at the same time. In that situation, the second thread to attempt the edit will fail with a conflict error. 
```

Comment:
This part might be a bit too detailed for a FAQ entry:

> Multiple threads can also simultaneously update separate tables or separate subsets of the same table. Each subset is ~20,000-80,000 rows depending on the width of the rows.

While this is granularity at which we do locking (ColumnSegments) - that should be transparent for the user as long as the same row is not being updated. I would remove the sentence "Each subset is ~20,000-80,000 rows depending on the width of the rows." here and talk only about the optimistic concurrency control.